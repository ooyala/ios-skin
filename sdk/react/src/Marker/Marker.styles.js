// @flow

import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  icon: {
    bottom: 4,
    height: 24,
    left: 4,
    position: 'absolute',
    width: 24,
  },
  root: {
    bottom: 0,
    minWidth: 10,
    position: 'absolute',
    top: 0,
  },
  text: {
    backgroundColor: 'red',
    bottom: 4,
    color: 'white',
    left: 4,
    position: 'absolute',
    width: 100,
  },
});
