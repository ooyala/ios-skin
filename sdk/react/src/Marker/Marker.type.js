// @flow

type BaseMarkerType = {
  backgroundColor?: string,
  end?: number | 'end', // number of seconds or till the end
  hoverColor?: string,
  markerColor?: string,
  start: number | 'start', // number of seconds or from the start
  type: string,
};

type TextMarkerType = BaseMarkerType & {
  text?: string,
  type: 'text',
};

type IconMarkerType = BaseMarkerType & {
  iconUrl?: string,
  imageUrl?: string,
  type: 'icon',
};

export type MarkerType = IconMarkerType | TextMarkerType;
