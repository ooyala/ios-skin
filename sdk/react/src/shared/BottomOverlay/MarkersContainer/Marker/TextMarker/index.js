// @flow

import TextMarker from './TextMarker';

export default TextMarker;
