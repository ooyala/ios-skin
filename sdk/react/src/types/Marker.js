// @flow

export type BaseMarker = {
  backgroundColor?: string,
  end?: number | 'end', // number of seconds or till the end
  hoverColor?: string,
  markerColor?: string,
  start: number | 'start', // number of seconds or from the start
};

export type TextMarker = BaseMarker & {
  text?: string,
  type: 'text',
};

export type IconMarker = BaseMarker & {
  iconUrl?: string,
  imageUrl?: string,
  type: 'icon',
};

export type Marker = TextMarker | IconMarker;
