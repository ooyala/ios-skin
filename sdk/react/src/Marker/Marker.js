// @flow

import * as React from 'react';
import { View } from 'react-native';

import IconMarker from '../IconMarker';
import styles from './Marker.styles';
import TextMarker from '../TextMarker';
import type { Marker as MarkerType } from '../types/Marker';

type Props = {
  accentColor?: ?string,
  duration: number,
  marker: MarkerType,
  onSeek: number => void,
};

const Marker = ({
  accentColor, duration, marker, onSeek,
}: Props) => {
  if (marker.type === 'text' && !marker.text) {
    return null;
  }

  const start = (marker.start === 'start' ? 0 : marker.start);
  const style = {};

  style.left = `${start / duration * 100}%`;

  if (marker.end) {
    style.right = (marker.end === 'end' ? 0 : `${100 - marker.end / duration * 100}%`);
  }

  if (marker.markerColor || accentColor) {
    style.backgroundColor = marker.markerColor || accentColor;
  }

  let addOn = null;

  if (marker.type === 'text' && marker.text) {
    addOn = <TextMarker onSeek={() => onSeek(start)} text={marker.text} />;
  } else if (marker.type === 'icon') {
    addOn = <IconMarker iconUrl={marker.iconUrl} imageUrl={marker.imageUrl} onSeek={() => onSeek(start)} />;
  }

  return (
    <View style={[styles.root, style]}>
      {addOn}
    </View>
  );
};

Marker.defaultProps = {
  accentColor: undefined,
};

export default Marker;
