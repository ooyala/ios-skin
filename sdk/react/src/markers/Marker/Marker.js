// @flow

import * as React from 'react';

import IconMarker from '../IconMarker';
import TextMarker from '../TextMarker';
import type { Marker as MarkerType } from '../../types/Marker';

type Props = {
  accentColor?: ?string,
  duration: number,
  marker: MarkerType,
  onSeek: number => void,
};

const Marker = ({
  accentColor, duration, marker, onSeek,
}: Props) => {
  const start = (marker.start === 'start' ? 0 : marker.start);
  const left = `${start / duration * 100}%`;

  if (marker.type === 'text' && marker.text) {
    return (
      <TextMarker
        backgroundColor={marker.backgroundColor || accentColor}
        onSeek={() => onSeek(start)}
        style={{ left }}
        text={marker.text}
      />
    );
  }

  if (marker.type === 'icon') {
    return (
      <IconMarker
        backgroundColor={marker.backgroundColor}
        iconUrl={marker.iconUrl}
        imageUrl={marker.imageUrl}
        onSeek={() => onSeek(start)}
        style={{ left }}
        touchColor={marker.markerColor || accentColor}
      />
    );
  }

  return null;
};

Marker.defaultProps = {
  accentColor: undefined,
};

export default Marker;
