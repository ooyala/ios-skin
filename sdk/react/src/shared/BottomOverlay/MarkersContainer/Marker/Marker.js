// @flow

import * as React from 'react';
import type { ViewStyleProp } from 'react-native/Libraries/StyleSheet/StyleSheet';

import IconMarker from './IconMarker';
import TextMarker from './TextMarker';
import type { Marker as MarkerType } from '../../../../types/Markers';

type Props = {
  accentColor?: ?string,
  duration: number,
  marker: MarkerType,
  onSeek: number => void,
  onTouch: () => void,
  style?: ViewStyleProp,
};

const Marker = ({
  accentColor, duration, marker, onSeek, onTouch, style,
}: Props) => {
  const start = (marker.start === 'start' ? 0 : marker.start);
  const left = `${start / duration * 100}%`;

  if (marker.type === 'text' && marker.text) {
    return (
      <TextMarker
        backgroundColor={marker.backgroundColor || accentColor}
        onSeek={() => onSeek(start)}
        onTouch={onTouch}
        style={[
          style,
          { left },
        ]}
        text={marker.text}
      />
    );
  }

  if (marker.type === 'icon') {
    return (
      <IconMarker
        backgroundColor={marker.backgroundColor}
        iconUrl={marker.iconUrl}
        imageUrl={marker.imageUrl}
        onSeek={() => onSeek(start)}
        onTouch={onTouch}
        style={[
          style,
          { left },
        ]}
        touchColor={marker.markerColor || accentColor}
      />
    );
  }

  return null;
};

Marker.defaultProps = {
  accentColor: undefined,
  style: undefined,
};

export default Marker;
