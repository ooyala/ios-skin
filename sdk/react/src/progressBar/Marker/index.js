// @flow

import Marker from './Marker';
import type { MarkerType as _MarkerType } from './Marker.type';

export default Marker;
export type MarkerType = _MarkerType;
