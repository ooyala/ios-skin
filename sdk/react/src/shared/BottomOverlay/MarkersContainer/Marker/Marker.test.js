// @flow

import { shallow } from 'enzyme';
import React from 'react';

import Marker from './Marker';

describe('Marker', () => {
  it('renders marker of the text type matching snapshot', () => {
    const wrapper = shallow(
      <Marker
        duration={60}
        marker={{
          start: 'start',
          type: 'text',
          text: 'Hello, world!',
        }}
        onSeek={() => undefined}
        onTouch={() => undefined}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders marker of the icon type matching snapshot', () => {
    const wrapper = shallow(
      <Marker
        duration={60}
        marker={{
          iconUrl: 'http://example.com/icon.png',
          start: 'start',
          type: 'icon',
        }}
        onSeek={() => undefined}
        onTouch={() => undefined}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
