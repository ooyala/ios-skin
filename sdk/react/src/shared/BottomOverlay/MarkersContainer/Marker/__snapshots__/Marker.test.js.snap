// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marker renders marker of the icon type matching snapshot 1`] = `
<IconMarker
  containerWidth={100}
  iconUrl="http://example.com/icon.png"
  leftPosition={0}
  onSeek={[Function]}
  onTouch={[Function]}
/>
`;

exports[`Marker renders marker of the text type matching snapshot 1`] = `
<TextMarker
  containerWidth={100}
  leftPosition={0}
  onSeek={[Function]}
  onTouch={[Function]}
  text="Hello, world!"
/>
`;
